gio/src/auto/async_initable.rs

```diff
@@ -0,0 +1,109 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::AsyncResult;
+use crate::Cancellable;
+use glib::object::IsA;
+use glib::translate::*;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::pin::Pin;
+use std::ptr;
+
+glib::wrapper! {
+    #[doc(alias = "GAsyncInitable")]
+    pub struct AsyncInitable(Interface<ffi::GAsyncInitable, ffi::GAsyncInitableIface>);
+
+    match fn {
+        type_ => || ffi::g_async_initable_get_type(),
+    }
+}
+
+impl AsyncInitable {
+    pub const NONE: Option<&'static AsyncInitable> = None;
+}
+
+pub trait AsyncInitableExt: 'static {
+    #[doc(alias = "g_async_initable_init_async")]
+    fn init_async<P: FnOnce(Result<(), glib::Error>) + 'static>(
```

Comment:
This needs to be `unsafe` as you are not allowed to call it multiple times on an object. Also the future variant below

Reply:
If I mark `init_async` as unsafe in Gir.toml, not only `init_future` is not automatically marked as unsafe, but it fails to compile. This seems to be a bug in gir.